test/e2e/set/scyllacluster/scyllacluster_tls.go

```diff
@@ -0,0 +1,380 @@
+
+	f := framework.NewFramework("scyllacluster")
+
+	g.It("should setup and maintain up to date TLS certificates", func() {
+		if !utilfeature.DefaultMutableFeatureGate.Enabled(features.AutomaticTLSCertificates) {
+			g.Skip(fmt.Sprintf("Skipping because %q feature is disabled", features.AutomaticTLSCertificates))
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+		o.Expect(sc.Spec.Datacenter.Racks).To(o.HaveLen(1))
+		sc.Spec.Datacenter.Racks[0].Members = 1
+
+		framework.By("Creating an initial ScyllaCluster with a single node")
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(
+			ctx,
+			sc,
+			metav1.CreateOptions{
+				FieldManager: f.FieldManager(),
+			},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to rollout")
+		waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Filling the cluster with data")
+		di, err := NewDataInserter(ctx, f.KubeClient().CoreV1(), sc, 1)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer di.Close()
+
+		err = di.Insert()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, tc := range []struct {
+			domains  []string
+			replicas int
+		}{
+			{
+				domains:  nil,
+				replicas: 1,
+			},
+			{
+				domains:  []string{"foo.scylladb.com", "bar.scylladb.com"},
+				replicas: 1,
+			},
+			{
+				domains:  []string{"foo.scylladb.com", "bar.scylladb.com"},
+				replicas: 2,
+			},
+			{
+				domains:  []string{"foo.scylladb.com", "bar.scylladb.com"},
+				replicas: 1,
+			},
+			{
+				domains:  nil,
+				replicas: 1,
+			},
+		} {
+			tc := tc
+			func() {
+				framework.By("Scaling the ScyllaCluster to %d replicas setting domains to %q", tc.replicas, tc.domains)
+				sc.ManagedFields = nil
+				sc.ResourceVersion = ""
+				sc.Spec.Datacenter.Racks[0].Members = int32(tc.replicas)
+				sc.Spec.DNSDomains = tc.domains
+				scData, err := runtime.Encode(scheme.Codecs.LegacyCodec(scyllav1.GroupVersion), sc)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				// TODO: Use generated Apply method when our clients have it.
+				sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(ctx, sc.Name, types.ApplyPatchType, scData, metav1.PatchOptions{
+					FieldManager: f.FieldManager(),
+					Force:        pointer.Bool(true),
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(sc.Spec.Datacenter.Racks[0].Members).To(o.BeEquivalentTo(tc.replicas))
+				o.Expect(sc.Spec.DNSDomains).To(o.BeEquivalentTo(tc.domains))
+
+				framework.By("Waiting for the ScyllaCluster to rollout")
+				waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
```

Comment:
nit: test below use "L" in it's name to distinguish root level and inner context, this one also could avoid shadowing

Reply:
fixed